articles/history-of-nodejs/history-of-nodejs.md

```diff
@@ -0,0 +1,168 @@
+---
+layout: engineering-education
+status: publish
+published: true
+url: /engineering-education/history-of-nodejs/
+title: The History of Node.js
+description: Node.js is a runtime server environment that uses JavaScript on the server side and asynchronous programming. It is a free and open source technology that runs on various platforms (Mac OS X, Unix, Windows, etc.)
+author: jethro-magaji
+date: 2020-08-17T00:00:00-08:00
+topics: []
+excerpt_separator: <!--more-->
+images:
+
+  - url: /engineering-education/history-of-nodejs/hero.jpg
+    alt:
+---
+Today we are going to talk about a revolutionary technology that is being used to build great web apps, it is called  *Node.js*.
+You may have probably heard of it or worked with it(or have been curious about it in the past), either way you are welcome to join us as we further explore the History of Node.js in this article. We are going to talk about the history, some of the basics and a few use cases examples that use this revolutionary technology.
+<!--more-->
+![Nodejs image](https://drive.google.com/uc?export=view&id=1ChxgpCW0JTXiGTeV15QnuVHv-byDCLuO)
+[source](https://www.simform.com/nodejs-use-case/)
+
+### What is Node.js?
+![Nodejs logo](https://drive.google.com/uc?export=view&id=13udtjL4Xz5YICREzII6hcHiv0xM8a_ds)
+[source](https://www.w3schools.com/nodejs/nodejs_intro.asp)
+
+Node.js is a runtime server environment that uses JavaScript on the server side and [asynchronous programming!](https://www.w3schools.com/nodejs/nodejs_intro.asp). It is a free and open source technology that runs on various platforms (Mac OS X, Unix, Windows, etc.)
+
+- **Node.js can:**
+    - create, open, read, write, delete, and close files on the server
+    - generate the dynamic page content
+    - collect form data
+    - add, delete, change data in your database
+
+- **Features**
+    - Easy to learn
+    - Fast
+    - No Buffering
+    - Consistent source code
+    - Massive Library support
+
+### The Early Beginnings
+Believe it or not, Node.js is only twelve years old. In comparison, JavaScript is 26 years old and the Web is 32 years old.
+[Ryan Dahl wrote Node.js](https://en.wikipedia.org/wiki/Node.js) about thirteen ago, in 2009. At first Node.js supported only Linux and Mac OS X. Dahl led the development and maintenance and later on it was [sponsored by Joyent](https://en.wikipedia.org/wiki/Node.js).
+The most popular web server of the time (Apache) was criticized back in 2009 due to its limited possibilities, because the Apache HTTP Server had to handle various connections concurrently(up to 10,000 or more) which were beyond its capabilities.
+The way of creating code was commonly through sequential programming, when there was either a blocked code in the entire process or an implied multiple execution stacks in cases of simultaneous connections.
```

Comment:
line 46 & 47 need some rewording - a bit difficult to read through or understand.  I think its trying to explain why apache has difficulties or was not the best option. Read lines above it to get a better sense of the flow - of my suggested edits so far.